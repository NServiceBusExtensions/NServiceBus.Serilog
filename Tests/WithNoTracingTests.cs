using System;
using System.Threading;
using System.Threading.Tasks;
using ApprovalTests;
using NServiceBus;
using Xunit;
using Xunit.Abstractions;

public class WithNoTracingTests : TestBase
{
    static ManualResetEvent resetEvent;
    static Exception exception;

    [Fact]
    public async Task Handler()
    {
        resetEvent = new ManualResetEvent(false);
        var configuration = new EndpointConfiguration("WithNoTracingTests");
        configuration.SendFailedMessagesTo("error");
        configuration.UsePersistence<InMemoryPersistence>();
        configuration.PurgeOnStartup(true);
        configuration.UseTransport<LearningTransport>();
        configuration.DisableRetries();

        configuration.Notifications.Errors.MessageSentToErrorQueue +=
            (sender, retry) =>
            {
                exception = retry.Exception;
                resetEvent.Set();
            };

        var endpoint = await Endpoint.Start(configuration);
        await endpoint.SendLocal(new StartHandler());
        if (!resetEvent.WaitOne(TimeSpan.FromSeconds(2)))
        {
            throw new Exception("No Set received.");
        }

        await endpoint.Stop();
        Approvals.Verify(exception.Message);
    }

    public WithNoTracingTests(ITestOutputHelper output) : base(output)
    {
    }
}