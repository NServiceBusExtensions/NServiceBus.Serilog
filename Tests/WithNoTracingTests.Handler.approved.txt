Expected to find a ILogger in the pipeline context. It is possible a call to SerilogTracingExtensions.EnableSerilogTracing is missing.